collector/diskstats_freebsd.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !nodiskstats
+
+package collector
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/shirou/gopsutil/disk"
```

Comment:
I'm not sure I'm a fan of this library. It pulls in a lot of stuff completely unrelated to what we're implementing here.

For example, pulling this in pulls in Windows WMI subsystem code.

Reply:
Would it be preferable to just use the same method the gopsutil package uses to get the data, and skip the package?

Reply:
Yes, the unix syscall method seems fine. We use it directly in collectors like `cpu_freebsd`.

It seems like the library has a lot of little helper functions in the common package of the code. They seem like a lot of simple stuff, like reproducing `strconv` functions while ignoring errors.

Possibly it's a build flag problem, I'm not an expert in these things.

Reply:
Alright I rewrote this without using the gopsutil package. Going to close this request and open a new one as the changes are very significant.